tfx_addons/sampling/executor.py

```diff
@@ -0,0 +1,286 @@
+# Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Executor for Sampler component."""
+
+import os
+import random
+from typing import Any, Dict, List, Text
+
+import apache_beam as beam
+import tensorflow as tf
+from tfx import types
+from tfx.dsl.components.base import base_beam_executor
+from tfx.dsl.io import fileio
+from tfx.types import artifact_utils
+from tfx.utils import io_utils, json_utils
+
+from tfx_addons.sampling import spec
+
+
+class Executor(base_beam_executor.BaseBeamExecutor):
+  """Executor for Sampler."""
+  def _CreatePipeline(self, unused_transform_output: Text) -> beam.Pipeline:
+    """Creates beam pipeline.
+    Args:
+      unused_transform_output: unused.
+    Returns:
+      Beam pipeline.
+    """
+
+    return self._make_beam_pipeline()
```

Comment:
Why do we need this?

Reply:
Not too sure - I believe Ross added that.